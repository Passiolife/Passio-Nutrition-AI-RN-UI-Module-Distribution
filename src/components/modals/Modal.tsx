import React from 'react';
import RNModal, { type ModalProps } from 'react-native-modal';

interface Props extends ModalProps {}

export const Modal: React.FC<Props> = ({ children, ...rest }) => {
  return <RNModal {...rest}>{children}</RNModal>;
};
