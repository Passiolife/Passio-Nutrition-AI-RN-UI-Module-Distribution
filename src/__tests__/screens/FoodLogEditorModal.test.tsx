// @ts-nocheck
import React from 'react';
import { MockProviders } from '../provider/MockProviders';
import { render } from '@testing-library/react-native';
import { FoodLogEditorModal } from '../../modals';
import { act } from 'react-test-renderer';
import type { PassioIDAttributes } from '@passiolife/nutritionai-react-native-sdk-v3/src/sdk/v2';

jest.mock('@passiolife/nutritionai-react-native-sdk-v3', () => {
  const mockSDK = {
    getAttributesForPassioID(): Promise<PassioIDAttributes | null> {
      return require('../assets/json/mock_passio_sdk_attribute.json');
    },
  };
  const mockPassioIcon = require('../utils/MockPassioIcon').default;

  return {
    PassioSDK: mockSDK,
    PassioIconView: mockPassioIcon,
    IconSize: 'PX90',
  };
});

describe('Food Log Editor Modal', () => {
  it('Renders correctly', () => {
    const onCancelPress = jest.fn();
    const onSavePress = jest.fn();

    const tree = render(
      <MockProviders>
        <FoodLogEditorModal
          foodLog={require('../assets/json/foodLog.json')}
          isVisible={true}
          onDismissFoodLogModal={onCancelPress}
          onSavedLog={onSavePress}
        />
      </MockProviders>
    );

    expect(tree).toMatchSnapshot();
  });

  it('checked nutrients calories, carbs, proteins and fat matched', () => {
    const onCancelPress = jest.fn();
    const onSavePress = jest.fn();

    const tree = render(
      <MockProviders>
        <FoodLogEditorModal
          foodLog={require('../assets/json/foodLog.json')}
          isVisible={true}
          onDismissFoodLogModal={onCancelPress}
          onSavedLog={onSavePress}
        />
      </MockProviders>
    );

    const calories = tree.getByTestId('testNutrientCalories');
    const carbs = tree.getByTestId('testNutrientCarbs');
    const protein = tree.getByTestId('testNutrientProtein');
    const fat = tree.getByTestId('testNutrientFat');
    expect(calories.props.children).toEqual(150);
    expect(carbs.props.children).toEqual([35, ' g']);
    expect(protein.props.children).toEqual([33, ' g']);
    expect(fat.props.children).toEqual([12, ' g']);
  });

  it('changed progress to checked that they affected to calories, carbs, proteins and fat of nutrients', () => {
    const onCancelPress = jest.fn();
    const onSavePress = jest.fn();
    const tree = render(
      <MockProviders>
        <FoodLogEditorModal
          foodLog={require('../assets/json/foodLog.json')}
          isVisible={true}
          onDismissFoodLogModal={onCancelPress}
          onSavedLog={onSavePress}
        />
      </MockProviders>
    );
    const textInput = tree.getByTestId('testTextInputOfSelectedQuantity');
    act(() => {
      textInput.props.onChangeText('2');
    });

    const calories = tree.getByTestId('testNutrientCalories');
    const carbs = tree.getByTestId('testNutrientCarbs');
    const protein = tree.getByTestId('testNutrientProtein');
    const fat = tree.getByTestId('testNutrientFat');

    expect(calories.props.children).toEqual(300);
    expect(carbs.props.children).toEqual([70, ' g']);
    expect(protein.props.children).toEqual([66, ' g']);
    expect(fat.props.children).toEqual([24, ' g']);
  });
});
