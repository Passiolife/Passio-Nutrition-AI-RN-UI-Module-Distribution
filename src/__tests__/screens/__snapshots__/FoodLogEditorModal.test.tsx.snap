// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Food Log Editor Modal Renders correctly 1`] = `
<Modal
  animated={true}
  animationType="slide"
  hardwareAccelerated={false}
  onRequestClose={[MockFunction]}
  transparent={true}
  visible={true}
>
  <View
    style={
      {
        "alignSelf": "center",
        "backgroundColor": "rgba(255,255,255,0.80)",
        "borderRadius": 16,
        "marginHorizontal": 50,
        "marginTop": 80,
        "padding": 4,
        "position": "absolute",
        "width": 734,
        "zIndex": 25,
      }
    }
  >
    <RCTScrollView
      contentContainerStyle={
        {
          "paddingBottom": 40,
        }
      }
      showsVerticalScrollIndicator={false}
      style={
        {
          "maxHeight": NaN,
        }
      }
    >
      <View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              [
                {
                  "backgroundColor": "#FFFFFF",
                  "borderRadius": 16,
                  "elevation": 4,
                  "shadowColor": "#00000029",
                  "shadowOffset": {
                    "height": 2.5,
                    "width": 1.5,
                  },
                  "shadowOpacity": 1,
                  "shadowRadius": 2.85,
                },
                {
                  "marginTop": 12,
                  "padding": 10,
                  "paddingBottom": 20,
                },
              ]
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  {
                    "alignSelf": "center",
                    "borderRadius": 24,
                    "marginLeft": 4,
                    "overflow": "hidden",
                    "width": 50,
                  }
                }
              >
                <Text>
                  mockPassioIcon
                </Text>
              </View>
              <View
                style={
                  {
                    "flex": 1,
                    "marginLeft": 10,
                  }
                }
              >
                <Text
                  ellipsizeMode="tail"
                  numberOfLines={2}
                  style={
                    {
                      "color": "#286CE2",
                      "fontSize": 20,
                      "fontWeight": "400",
                      "lineHeight": 24,
                      "textTransform": "capitalize",
                    }
                  }
                >
                  orange tomatoes
                </Text>
                <Text
                  style={
                    {
                      "color": "#333333",
                      "fontSize": 13,
                      "fontWeight": "400",
                      "lineHeight": 16,
                      "marginTop": 2,
                      "textTransform": "capitalize",
                    }
                  }
                >
                  1 serving (100g)
                </Text>
              </View>
            </View>
            <View
              style={
                {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginTop": 15,
                }
              }
            >
              <View
                style={
                  {
                    "alignItems": "center",
                  }
                }
              >
                <Text
                  style={
                    {
                      "color": "#333333",
                      "fontSize": 13,
                      "fontWeight": "600",
                      "lineHeight": 16,
                      "textTransform": "capitalize",
                    }
                  }
                >
                  Calories
                </Text>
                <Text
                  style={
                    {
                      "color": "#333333",
                      "fontSize": 24,
                      "fontWeight": "300",
                      "lineHeight": 29,
                    }
                  }
                  testID="testNutrientCalories"
                >
                  150
                </Text>
              </View>
              <View
                style={
                  {
                    "alignItems": "center",
                  }
                }
              >
                <Text
                  style={
                    {
                      "color": "#333333",
                      "fontSize": 13,
                      "fontWeight": "600",
                      "lineHeight": 16,
                      "textTransform": "capitalize",
                    }
                  }
                >
                  Carbs
                </Text>
                <Text
                  style={
                    {
                      "color": "#333333",
                      "fontSize": 24,
                      "fontWeight": "300",
                      "lineHeight": 29,
                    }
                  }
                  testID="testNutrientCarbs"
                >
                  35
                   g
                </Text>
              </View>
              <View
                style={
                  {
                    "alignItems": "center",
                  }
                }
              >
                <Text
                  style={
                    {
                      "color": "#333333",
                      "fontSize": 13,
                      "fontWeight": "600",
                      "lineHeight": 16,
                      "textTransform": "capitalize",
                    }
                  }
                >
                  Protein
                </Text>
                <Text
                  style={
                    {
                      "color": "#333333",
                      "fontSize": 24,
                      "fontWeight": "300",
                      "lineHeight": 29,
                    }
                  }
                  testID="testNutrientProtein"
                >
                  33
                   g
                </Text>
              </View>
              <View
                style={
                  {
                    "alignItems": "center",
                  }
                }
              >
                <Text
                  style={
                    {
                      "color": "#333333",
                      "fontSize": 13,
                      "fontWeight": "600",
                      "lineHeight": 16,
                      "textTransform": "capitalize",
                    }
                  }
                >
                  Fat
                </Text>
                <Text
                  style={
                    {
                      "color": "#333333",
                      "fontSize": 24,
                      "fontWeight": "300",
                      "lineHeight": 29,
                    }
                  }
                  testID="testNutrientFat"
                >
                  12
                   g
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            [
              {
                "backgroundColor": "#FFFFFF",
                "borderRadius": 16,
                "elevation": 4,
                "shadowColor": "#00000029",
                "shadowOffset": {
                  "height": 2.5,
                  "width": 1.5,
                },
                "shadowOpacity": 1,
                "shadowRadius": 2.85,
              },
              {
                "flexDirection": "column",
                "marginTop": 8,
                "paddingBottom": 16,
                "paddingTop": 8,
              },
            ]
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              style={
                {
                  "color": "#333333",
                  "fontSize": 13,
                  "fontWeight": "600",
                  "lineHeight": 16,
                  "paddingHorizontal": 16,
                }
              }
            >
              Edit Amount
            </Text>
          </View>
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "flexWrap": "nowrap",
                "justifyContent": "flex-start",
                "marginHorizontal": 16,
                "marginTop": 10,
              }
            }
          >
            <View>
              <TextInput
                keyboardType="numeric"
                onChangeText={[Function]}
                style={
                  [
                    {
                      "backgroundColor": "#EFF0F6",
                      "borderColor": "#D9DBE9",
                      "borderRadius": 10,
                      "borderWidth": 1,
                      "height": 40,
                      "paddingHorizontal": 8,
                    },
                    {},
                    {
                      "minWidth": 50,
                      "textAlign": "right",
                    },
                  ]
                }
                testID="testTextInputOfSelectedQuantity"
                value="1"
              />
            </View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={false}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "backgroundColor": "#286CE2",
                  "borderRadius": 25,
                  "justifyContent": "center",
                  "marginStart": 16,
                  "maxWidth": 150,
                  "opacity": 1,
                  "paddingHorizontal": 16,
                  "paddingVertical": 8,
                }
              }
            >
              <Text
                style={
                  {
                    "color": "#FFFFFF",
                    "fontSize": 14,
                    "textTransform": "capitalize",
                  }
                }
              >
                serving
              </Text>
            </View>
            <Text
              style={
                {
                  "color": "#333333",
                  "fontSize": 15,
                  "marginStart": 16,
                }
              }
            >
              (100 g )
            </Text>
          </View>
          <View
            style={
              {
                "marginHorizontal": 16,
                "marginTop": 10,
              }
            }
          >
            <RNCSlider
              disabled={false}
              inverted={false}
              lowerLimit={-9007199254740991}
              maximumTrackTintColor="#CCCCCC"
              maximumValue={5}
              minimumTrackTintColor="#286CE2"
              minimumValue={0}
              onChange={[Function]}
              onRNCSliderAccessibilityAction={null}
              onRNCSliderSlidingComplete={null}
              onRNCSliderSlidingStart={null}
              onRNCSliderValueChange={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              step={0.5}
              style={
                {
                  "height": 40,
                }
              }
              tapToSeek={false}
              thumbTintColor="#286CE2"
              upperLimit={9007199254740991}
              value={1}
            />
          </View>
          <View
            style={
              {
                "marginTop": 10,
              }
            }
          >
            <RCTScrollView
              contentContainerStyle={
                {
                  "flexDirection": "row",
                }
              }
              data={
                [
                  {
                    "quantity": 1,
                    "unit": "serving",
                  },
                ]
              }
              getItem={[Function]}
              getItemCount={[Function]}
              horizontal={true}
              keyExtractor={[Function]}
              onContentSizeChange={[Function]}
              onLayout={[Function]}
              onMomentumScrollBegin={[Function]}
              onMomentumScrollEnd={[Function]}
              onScroll={[Function]}
              onScrollBeginDrag={[Function]}
              onScrollEndDrag={[Function]}
              removeClippedSubviews={false}
              renderItem={[Function]}
              scrollEventThrottle={0.0001}
              showsHorizontalScrollIndicator={false}
              stickyHeaderIndices={[]}
              viewabilityConfigCallbackPairs={[]}
            >
              <View>
                <View
                  onFocusCapture={[Function]}
                  onLayout={[Function]}
                  style={
                    [
                      {
                        "flexDirection": "row",
                      },
                      null,
                    ]
                  }
                >
                  <View
                    style={
                      [
                        {
                          "backgroundColor": "#FFFFFF",
                          "borderRadius": 16,
                          "elevation": 4,
                          "shadowColor": "#00000029",
                          "shadowOffset": {
                            "height": 2.5,
                            "width": 1.5,
                          },
                          "shadowOpacity": 1,
                          "shadowRadius": 2.85,
                        },
                        [
                          {
                            "flexDirection": "column",
                            "marginBottom": 8,
                            "marginLeft": 8,
                            "marginTop": 8,
                          },
                          {
                            "backgroundColor": "#123456",
                          },
                        ],
                      ]
                    }
                  >
                    <View
                      accessibilityState={
                        {
                          "busy": undefined,
                          "checked": undefined,
                          "disabled": undefined,
                          "expanded": undefined,
                          "selected": undefined,
                        }
                      }
                      accessibilityValue={
                        {
                          "max": undefined,
                          "min": undefined,
                          "now": undefined,
                          "text": undefined,
                        }
                      }
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        {
                          "opacity": 1,
                          "overflow": "hidden",
                        }
                      }
                    >
                      <Text
                        style={
                          [
                            {
                              "fontSize": 14,
                              "paddingHorizontal": 15,
                              "paddingVertical": 7,
                            },
                            {
                              "color": "#FFFFFF",
                            },
                          ]
                        }
                      >
                        1 serving 
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </RCTScrollView>
          </View>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              [
                {
                  "backgroundColor": "#FFFFFF",
                  "borderRadius": 16,
                  "elevation": 4,
                  "shadowColor": "#00000029",
                  "shadowOffset": {
                    "height": 2.5,
                    "width": 1.5,
                  },
                  "shadowOpacity": 1,
                  "shadowRadius": 2.85,
                },
                {
                  "flexDirection": "column",
                  "marginTop": 8,
                  "paddingBottom": 16,
                  "paddingTop": 8,
                },
              ]
            }
          >
            <Text
              style={
                {
                  "color": "#333333",
                  "fontSize": 13,
                  "fontWeight": "600",
                  "lineHeight": 16,
                  "paddingHorizontal": 16,
                }
              }
            >
              Meal Time
            </Text>
            <View
              style={
                {
                  "alignSelf": "center",
                  "paddingHorizontal": 16,
                  "paddingTop": 15,
                }
              }
            >
              <View
                style={
                  {
                    "flexDirection": "row",
                    "height": 30,
                  }
                }
              >
                <View
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    [
                      {
                        "alignItems": "center",
                        "backgroundColor": "#EEEEEE",
                        "justifyContent": "center",
                        "marginRight": 1,
                        "width": 78,
                      },
                      {
                        "borderBottomLeftRadius": 21,
                        "borderTopLeftRadius": 21,
                      },
                      false,
                      false,
                    ]
                  }
                >
                  <Text
                    style={
                      [
                        {
                          "color": "#333333",
                          "fontSize": 13,
                        },
                        false,
                      ]
                    }
                  >
                    Breakfast
                  </Text>
                </View>
                <View
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    [
                      {
                        "alignItems": "center",
                        "backgroundColor": "#EEEEEE",
                        "justifyContent": "center",
                        "marginRight": 1,
                        "width": 78,
                      },
                      false,
                      false,
                      false,
                    ]
                  }
                >
                  <Text
                    style={
                      [
                        {
                          "color": "#333333",
                          "fontSize": 13,
                        },
                        false,
                      ]
                    }
                  >
                    Lunch
                  </Text>
                </View>
                <View
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    [
                      {
                        "alignItems": "center",
                        "backgroundColor": "#EEEEEE",
                        "justifyContent": "center",
                        "marginRight": 1,
                        "width": 78,
                      },
                      false,
                      false,
                      {
                        "backgroundColor": "#286CE2",
                      },
                    ]
                  }
                >
                  <Text
                    style={
                      [
                        {
                          "color": "#333333",
                          "fontSize": 13,
                        },
                        {
                          "color": "#FFFFFF",
                        },
                      ]
                    }
                  >
                    Dinner
                  </Text>
                </View>
                <View
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    [
                      {
                        "alignItems": "center",
                        "backgroundColor": "#EEEEEE",
                        "justifyContent": "center",
                        "marginRight": 1,
                        "width": 78,
                      },
                      false,
                      {
                        "borderBottomRightRadius": 21,
                        "borderTopRightRadius": 21,
                        "marginRight": 0,
                      },
                      false,
                    ]
                  }
                >
                  <Text
                    style={
                      [
                        {
                          "color": "#333333",
                          "fontSize": 13,
                        },
                        false,
                      ]
                    }
                  >
                    Snack
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              [
                {
                  "backgroundColor": "#FFFFFF",
                  "borderRadius": 16,
                  "elevation": 4,
                  "shadowColor": "#00000029",
                  "shadowOffset": {
                    "height": 2.5,
                    "width": 1.5,
                  },
                  "shadowOpacity": 1,
                  "shadowRadius": 2.85,
                },
                {
                  "flexDirection": "column",
                  "marginTop": 8,
                  "paddingBottom": 16,
                  "paddingTop": 8,
                },
              ]
            }
          >
            <Text
              style={
                {
                  "color": "#333333",
                  "fontSize": 13,
                  "fontWeight": "600",
                  "lineHeight": 16,
                  "paddingHorizontal": 16,
                }
              }
            >
              Timestamp
            </Text>
            <View
              style={
                {
                  "paddingHorizontal": 19,
                  "paddingTop": 15,
                }
              }
            >
              <Text
                style={
                  {
                    "alignSelf": "center",
                    "color": "#286CE2",
                    "fontSize": 15,
                    "fontWeight": "400",
                    "lineHeight": 18,
                  }
                }
              >
                Mon, Oct 4th 2021,  1:14 pm
              </Text>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-evenly",
          "marginHorizontal": 32,
          "marginVertical": 16,
        }
      }
    >
      <View
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "justifyContent": "center",
              "minWidth": 100,
              "paddingHorizontal": 16,
            },
            {
              "borderRadius": 20,
              "height": 30,
            },
            {
              "backgroundColor": "transparent",
              "borderColor": "#123456",
              "borderWidth": 1,
            },
            {
              "borderRadius": 14,
              "flex": 1,
              "justifyContent": "center",
              "marginHorizontal": 8,
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "fontWeight": "400",
              },
              {
                "fontSize": 15,
                "lineHeight": 18,
              },
              {
                "color": "#123456",
              },
            ]
          }
        >
          Cancel
        </Text>
      </View>
      <View
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "justifyContent": "center",
              "minWidth": 100,
              "paddingHorizontal": 16,
            },
            {
              "borderRadius": 20,
              "height": 30,
            },
            {
              "backgroundColor": "#123456",
              "borderColor": "#123456",
              "borderWidth": 0,
            },
            {
              "borderRadius": 14,
              "flex": 1,
              "justifyContent": "center",
              "marginHorizontal": 8,
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "fontWeight": "400",
              },
              {
                "fontSize": 15,
                "lineHeight": 18,
              },
              {
                "color": "#FFFFFF",
              },
            ]
          }
        >
          Save
        </Text>
      </View>
    </View>
  </View>
</Modal>
`;
